R/webshot.R

```diff
@@ -139,18 +162,28 @@ webshot <- function(
     }
   }
 
+  data <- data.frame(url = url, file = file, vwidth = vwidth, vheight = vheight)
+
+  # Params selector and expand can be either a vector that need to be concatenated
+  # or a list of such vectors. This function can be used to convert them into a
+  # character vector with the desired format.
+  argToVec <- function(arg) {
+    if (!is.list(arg)) return(paste(arg, collapse = ","))
+    sapply(arg, function(x) {
+      if (is.null(x) || is.na(x)) return(NA_character_)
+      else paste(x, collapse = ",")
+    })
+  }
+
+  if (!is.null(selector)) data$selector <- argToVec(selector)
```

Comment:
`selector` and `expand` should probably check for the correct number of items (either 1 or 4) in the R code instead of in the Javascript, because an error in the JS could now result in a partially-successful run. Better to have it not work at all than to partially work.